### PR TITLE
[WIP] The 'Leaf: Machine Learning for hackers' book

### PR DESCRIPTION
**Pull request by MichaelHirn**
This PR is a work in progress. So far I added the 1. Leaf and 2. Layers chapters.

Feedback for the overall structure of the book as well as the general ideas and style of the first two chapters are highly welcome.

There are probably a lot of typos and grammar mistakes in there. You don't have to bother yet pointing them out, the text might change quickly.

REFERENCE: #91

### CODE REVIEW

**Comment by MichaelHirn** [comment]
Build instructions, if you want to see the final HTML version (requires [mdbook](https://github.com/azerupi/mdBook))

```bash
cargo install mdbook
~/.cargo/bin/mdbook build ./doc
# open the doc dir in your browser
```

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f25d5b3bde51ff68ea36ce930b3f09c5fd12ae9b
  * Your commit message body contains a line that is longer than 100 characters

Guidelines are available at https://github.com/autumnai/leaf/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f25d5b3bde51ff68ea36ce930b3f09c5fd12ae9b
  * Your commit message body contains a line that is longer than 100 characters

Guidelines are available at https://github.com/autumnai/leaf/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by MichaelHirn** [comment]
The **Leaf: Machine Learning for Hackers** book is ready for review, I think.

Feedback, improvement ideas and especially PRs are highly welcome ;)

http://autumnai.com/leaf/book/

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: cd1d476cd92fd36498f3cb11caadb9aa611b57ba
  * Your commit message body contains a line that is longer than 100 characters

Guidelines are available at https://github.com/autumnai/leaf/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by MichaelHirn** [comment]
@homu  r+

File: doc/book/backend.html [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                        <i id="print-button" class="fa fa-print" title="Print this book"></i>
+                    </div>
+                </div>
+
+                <div id="content" class="content">
+                    <h1>Backend</h1>
+<p>Via the concept of a backend we can abstract over the platform we will execute
+or optimize a network on. The construction of a backend is trivial. The backend
+is passed to the <code>Solver</code>, (one backend for <code>network</code> and one for the
+<code>objectve</code>). The Solver than executes all operations on the provided backend.</p>
```
Comment (jeanm) [inline doc/book/backend.html]: The solver *then* executes

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: cd1d476cd92fd36498f3cb11caadb9aa611b57ba
  * Your commit message body contains a line that is longer than 100 characters

Guidelines are available at https://github.com/autumnai/leaf/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: cd1d476cd92fd36498f3cb11caadb9aa611b57ba
  * Your commit message body contains a line that is longer than 100 characters

Guidelines are available at https://github.com/autumnai/leaf/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by MichaelHirn** [comment]
@homu r+

**Pull request merged**